src/openrct2/ride/Vehicle.cpp

```diff
@@ -3598,6 +3600,23 @@ void Vehicle::UpdateCollisionSetup()
 
         train->sub_state = 2;
 
+#ifdef ENABLE_SCRIPTING
```

Comment:
all of the other crash hooks only do the train head. Why for this one do you want all of the train cars?

Reply:
Unless if I'm misunderstanding how the Vehicle class works, Vehicle::CrashOnLand() and Vehicle::CrashOnWater() is called for each individual car in a train, since the cars can hit the ground at different times. Vehicle::UpdateCollisionSetup() appears to be called for only the car that directly collided with a car from another train, hence the for loop that goes over every car. All three hook calls should fire for all cars in a train, not just the head.

Reply:
Oh yeah good point